### PR TITLE
refactor: payment network factory

### PR DESCRIPTION
**Pull request by benjlevesque**
# Description
The PaymentNetworkFactory was used as a set of static methods. I refactored it to be able to pass common dependencies in the constructor, like in the rest of the codebase.

# Context
This is in preparation for additional properties available for Payment Detection customization, like Subgraph URLs.
The idea is to be able to pass the configuration to the request client directly

### CODE REVIEW

File: packages/payment-detection/src/payment-network-factory.ts [hunk 1]
```diff
@@ -62,133 +58,84 @@ const anyCurrencyPaymentNetwork: IPaymentNetworkModuleByType = {
...
-    return { ...paymentNetwork, ...anyCurrencyPaymentNetwork };
-  }
-
-  /**
-   * Checks if a networkId is part of the supported networks for given currency
-   *
-   * @param paymentNetworkId The networkId to check is supported by this currency
-   * @param currency The currency to check the supported networks for
-   */
-  public static currencySupportsPaymentNetwork(
```
Comment (benjlevesque) [inline packages/payment-detection/src/payment-network-factory.ts:187]: removed unused method

File: packages/payment-detection/src/payment-network-factory.ts [hunk 2]
```diff
@@ -62,133 +58,84 @@ const anyCurrencyPaymentNetwork: IPaymentNetworkModuleByType = {
...
-      currencyManager,
-    });
-  }
-
-  /**
-   * Gets the payment networks supported for a Currency object
-   *
-   * @param currency The currency to get the supported networks for
-   */
-  public static supportedPaymentNetworksForCurrency(
```
Comment (benjlevesque) [inline packages/payment-detection/src/payment-network-factory.ts:167]: remove unnecessary abstraction

File: packages/request-client.js/src/http-request-network.ts [hunk 3]
```diff
@@ -6,19 +6,16 @@ import {
...
 import HttpDataAccess from './http-data-access';
-import HttpMetaMaskDataAccess from './http-metamask-data-access';
 import MockDataAccess from './mock-data-access';
 import MockStorage from './mock-storage';
 
 /**
  * Exposes RequestNetwork module configured to use http-data-access.
  */
 export default class HttpRequestNetwork extends RequestNetwork {
-  /** Public for test purpose */
```
Comment (benjlevesque) [inline packages/request-client.js/src/http-request-network.ts:19]: use proper mocks

File: packages/integration-test/test/node-client.test.ts [hunk 4]
```diff
@@ -415,7 +415,7 @@ describe('Request client using a request node', () => {
 
     await fetchedRequest.refresh();
     expect(fetchedRequest.getData().expectedAmount).toBe('0');
-  });
+  }, 60000);
```
Comment (benjlevesque) [inline packages/integration-test/test/node-client.test.ts:418]: this test fails sometimes

File: packages/request-client.js/src/api/request-network.ts [hunk 5]
```diff
@@ -44,29 +43,31 @@ export default class RequestNetwork {
...
     decryptionProvider,
-    bitcoinDetectionProvider,
     currencyManager,
+    paymentOptions,
   }: {
     dataAccess: DataAccessTypes.IDataAccess;
     signatureProvider?: SignatureProviderTypes.ISignatureProvider;
     decryptionProvider?: DecryptionProviderTypes.IDecryptionProvider;
-    bitcoinDetectionProvider?: PaymentTypes.IBitcoinDetectionProvider;
-    currencies?: CurrencyInput[];
```
Comment (benjlevesque) [inline packages/request-client.js/src/api/request-network.ts:61]: removed unused

File: packages/integration-test/test/native-token.test.ts [hunk 6]
```diff
@@ -5,42 +5,34 @@ import { AdvancedLogic } from '@requestnetwork/advanced-logic';
...
+    const paymentNetwork = paymentNetworkFactory.createPaymentNetwork(
+      PaymentTypes.PAYMENT_NETWORK_ID.NATIVE_TOKEN,
+      RequestLogicTypes.CURRENCY.ETH,
+      'aurora-testnet',
+    );
     const action = await paymentNetwork.createExtensionsDataForCreation(createCreationActionParams);
     expect(action.parameters.paymentAddress).toEqual('payment.testnet');
     expect(action.parameters.paymentNetworkName).toEqual('aurora-testnet');
   });
   it('throws without a payment network name', async () => {
```
Comment (bertux) [inline packages/integration-test/test/native-token.test.ts:30]: for me it seems the same execution than the previous test so I don't see where the paymentNetworkName is removed from createCreationActionParams as it was before.

**Review by bertux — APPROVED** [review]

**Review by alexandre-abrioux — COMMENTED** [review]
Clean! 💯
I made a small fixup for the request-client HTML test file here: https://github.com/RequestNetwork/requestNetwork/pull/937

**Pull request merged**